app/(dashboard)/dashboard/settings/actions.ts

```diff
@@ -0,0 +1,17 @@
+"use server"
+
+import { getServerSession } from "next-auth"
+
+import { createApiKey } from "@/lib/apikey"
+import { authOptions } from "@/lib/auth"
+import { TApiKeyCreateInput } from "@/lib/types/apiKey"
+
+export async function createApiKeyAction(
+  accountId: string,
+  apiKeyData: TApiKeyCreateInput
+) {
+  const session = await getServerSession(authOptions)
+  if (!session) return
+
+  return await createApiKey(accountId, apiKeyData)
```

Comment:
We need an additional check here if the current user has access rights to the repository so he doesn't create an api key for a repo he doesn't belong to